packages/react-liveness/src/components/FaceLivenessDetector/shared/LivenessIconWithPopover.tsx

```diff
@@ -5,60 +5,74 @@
 
 export interface LivenessIconWithPopoverProps {
   children: string;
+  headingText: string;
+  labelText: string;
 }
 
-export const LivenessIconWithPopover: React.FC<LivenessIconWithPopoverProps> =
-  ({ children }) => {
-    const breakpoint = useThemeBreakpoint();
-    const [shouldShowPopover, setShouldShowPopover] = React.useState(false);
-    const wrapperRef = React.useRef<HTMLDivElement | null>(null);
-    const isMobileScreen = breakpoint === 'base';
+export const LivenessIconWithPopover: React.FC<
+  LivenessIconWithPopoverProps
+> = ({ children, headingText, labelText }) => {
+  const breakpoint = useThemeBreakpoint();
+  const [shouldShowPopover, setShouldShowPopover] = React.useState(false);
+  const wrapperRef = React.useRef<HTMLDivElement | null>(null);
+  const isMobileScreen = breakpoint === 'base';
 
-    React.useEffect(() => {
-      function handleClickOutside(event: MouseEvent) {
-        if (
-          shouldShowPopover &&
-          wrapperRef.current &&
-          !wrapperRef.current.contains(event.target as Node)
-        ) {
-          setShouldShowPopover(false);
-        }
+  React.useEffect(() => {
+    function handleClickOutside(event: MouseEvent) {
+      if (
+        shouldShowPopover &&
+        wrapperRef.current &&
+        !wrapperRef.current.contains(event.target as Node)
+      ) {
+        setShouldShowPopover(false);
       }
-      document.addEventListener('mousedown', handleClickOutside);
-      return () => {
-        document.removeEventListener('mousedown', handleClickOutside);
-      };
-    }, [wrapperRef, shouldShowPopover]);
+    }
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [wrapperRef, shouldShowPopover]);
 
-    return (
-      <Flex
-        className={LivenessClassNames.Popover}
+  return (
+    <div className={LivenessClassNames.Popover} ref={wrapperRef}>
+      <Button
+        aria-controls="photosensitivity-description"
+        aria-expanded={shouldShowPopover}
+        role="alertdialog"
+        aria-label={labelText}
+        aria-describedby="photosensitivity-description"
+        colorTheme="info"
+        id="popover-button"
         onClick={() => setShouldShowPopover(!shouldShowPopover)}
-        ref={wrapperRef}
         testId="popover-icon"
       >
         <AlertIcon ariaHidden variation="info" />
-        {shouldShowPopover && (
-          <>
-            <Flex className={LivenessClassNames.PopoverAnchor} />
-            <Flex className={LivenessClassNames.PopoverAnchorSecondary} />
-            <Flex
-              className={LivenessClassNames.PopoverContainer}
-              left={isMobileScreen ? -190 : -108}
-              data-testid="popover-text"
-            >
-              {children}
-            </Flex>
-          </>
-        )}
-      </Flex>
-    );
-  };
+      </Button>
+      {shouldShowPopover && (
```

Comment:
The more i look at this component I'm becoming a bit confused. If the outer wrapper `div` is styled as a `Popover` via the `LivenessClassNames.Popover` className why are we partially rendering the content?

Reply:
What do you mean by that? Isn't the point of a popover to be hidden until you click the button?